Mathlib/Data/List/GetD.lean

```diff
@@ -30,46 +30,48 @@ variable (d : α)
 theorem getD_eq_getElem {n : ℕ} (hn : n < l.length) : l.getD n d = l[n] := by
   grind
 
-theorem getD_map {n : ℕ} (f : α → β) : (map f l).getD n (f d) = f (l.getD n d) := by simp
+theorem getD_eq_getElem? (i : Fin l.length) : l.getD i d = l[i]?.get (by simp) := by
+  simp only [getD_eq_getElem?_getD, Fin.is_lt, getElem?_pos, Option.getD_some, Fin.getElem_fin,
+    Option.get_some]
```

Comment:
What is the point of this theorem?

Reply:
I meant: what problem are you solving by adding this lemma? I'd prefer that we don't just inflate the `List` API in Mathlib. If there's some other PR where you're running into needing this, I'd like to know about that. But I'm skeptical about just adding lemmas one at a time in mathlib to the `List` namespace.